client/blocks/conversation-follow-button/helper.js

```diff
@@ -0,0 +1,20 @@
+/*
+ * @format
+ */
+
+/**
+ * Internal dependencies
+ */
+import { isDiscoverPost } from 'reader/discover/helper';
+import { shouldShowComments } from 'blocks/comments/helper';
+import config from 'config';
+
+export function shouldShowConversationFollowButton( post ) {
```

Comment:
An alternative to this helper would be passing the full `post` object into `<ConversationFollowButton>` and having it make the decision on whether to display itself.